src/v/kafka/server/handlers/create_topics.cc

```diff
@@ -250,6 +262,13 @@ ss::future<response_ptr> create_topics_handler::handle(
           std::back_inserter(response.data.topics),
           [&ctx](const creatable_topic& t) {
               auto result = generate_successfull_result(t);
+              if (ctx.metadata_cache().contains(model::topic_namespace_view{
+                    model::kafka_namespace, t.name})) {
+                  result.error_code = error_code::topic_already_exists;
+                  return result;
+              }
+              result.num_partitions = t.num_partitions;
+              result.replication_factor = t.replication_factor;
```

Comment:
question: just to confirm, if the topic already exists, we do _not_ return that topics partition count or RF?

Reply:
Correct. That's what the latest Kafka does.